app/controllers/docker_manager/admin_controller.rb

```diff
@@ -87,16 +87,18 @@ def latest
     def upgrade
       repo = find_repos(params[:path])
       raise Discourse::NotFound unless repo.present?
+      script_path = File.expand_path(File.join(__dir__, '../../../scripts/docker_manager_upgrade.rb'))
 
-      pid = fork do
-        exit if fork
-        Process.setsid
-        exit if fork
-        Upgrader.new(current_user.id, repo, repo_version(repo)).upgrade
-      end
-
-      Process.waitpid(pid)
-
+      pid = spawn(
+        {
+          'UPGRADE_USER_ID' => current_user.id.to_s,
+          'UPGRADE_PATH' => params[:path].to_s,
+          'UPGRADE_REPO_VERSION' => repo_version(repo).to_s,
+          'RAILS_ENV' => Rails.env
+        },
+        "bundle exec rails runner #{script_path}"
+      )
+      Process.detach(pid)
```

Comment:
technically since this is double forking it could wait for the pid, a detach is also ok